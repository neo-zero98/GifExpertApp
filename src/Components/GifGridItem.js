import React from 'react'
import PropTypes  from "prop-types";

const GifGridItem = ({img}) => {
    return (
        <>
            <div className="card">
                <img src={img.url} />
                <p>{img.title}</p>
            </div>
        </>
    )
}

GifGridItem.protoTypes = {
    img: PropTypes.object.isRequired
}

export default GifGridItem
