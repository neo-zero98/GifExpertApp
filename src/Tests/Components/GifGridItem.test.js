import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../Components/GifGridItem";

describe('Pruebas en <GifGridItem />', () => {
    const img = {
        url: '',
        title: ''
    }
    const wrapper = shallow(<GifGridItem img={img} />); 
    test('debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    }); 

    test('Debe de tener un titulo con el parrafo', () => {
       const p = wrapper.find('p');
       expect(p.text().trim()).toBe(img.title);
    });

    test('Debe tener la imagen igual a la url y alt de los promps', () => {
        const imagen = wrapper.find('img');
        expect(imagen.prop('src')).toBe(img.url);
        // expect(imagen.prop('alt')).toBe(img.title);
    });

    test('debe de tener la clase card', () => {
       const div = wrapper.find('div'); 
       expect(div.prop('className')).toBe('card');
        // console.log(div.prop('className);
    });
    
    
    
});
