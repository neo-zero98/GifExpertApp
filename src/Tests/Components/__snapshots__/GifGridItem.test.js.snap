// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<Fragment>
  <div
    className="card"
  >
    <img
      src=""
    />
    <p />
  </div>
</Fragment>
`;
